docker-compose.yml

```diff
@@ -1,18 +1,24 @@
-version: "2.0"
+version: '3'
 services:
   titra:
     image: kromit/titra
-    container_name: titra
+    container_name: titra_app
     depends_on:
       - mongodb
     environment:
       - ROOT_URL=${ROOT_URL}
       - MONGO_URL=mongodb://mongodb/titra?directConnection=true
       - PORT=3000
     ports:
-      - "3000:3000"
+      - "127.0.0.1:3000:3000"
```

Comment:
Don't expose on any interfaces by default

Reply:
What is the reason why we should do this by default? I see the benefit if titra runs behind a reverse proxy but most people will use the docker compose file as a starting point to try titra on a server so it shouldn't be limited to localhost in my opinion. Please correct me if I am missing something but I think this is what this change would imply.

Reply:
This way it's secure by default/design and no exposed on unwanted networks. 

And if people want to expose it to another network, let's say on internet for a VPS, or on XYZ network on an enterprise network, they will just change `127.0.0.1` to whatever the IP address of the interface is. Not specifying an IP address will lead to the address `0.0.0.0` which means all addresses, which means the container will be listening at all interfaces, which may lead to unwanted exposure without the admin knowing it.

The security posture is to have people enabling whatever interface/IP address they need on a case-by-case basis rather that exposing anything anywhere by default.

Reply:
Let me run some tests to fully understand the implications of this - I will update the PR and get back to you with my results.

Reply:
You can leave out the `ports`, since the Docker images already define `EXPOSE` rules. Then other containers, and also the host system, can access a container via the internal bridge networking. It is not neccessary to expose the application locally, and if, only for development instances available to `::1`.

A production(-like) setup, even if only for self-hosted scenarios, should always bring a reverse proxy. A simple example could be provided with an additional Caddy container or working Traefik labels, for example.

Reply:
I agree for production environments, but the main purpose of this compose file was initially to get a "demo" environment up and running locally with a one-line command. I think the best solution would be to create two or more docker-compose files for every purpose as an example. Then on the other hand it is also hard to create a "production-ready" docker compose file because e.g. the traefik labels cannot be pre-defined for custom domains...